includes/parse_md.php

```diff
@@ -74,9 +74,13 @@ function parse_md($markdown) {
     $pd = new ParsedownExtra();
     $content = $pd->text($md);
 
-    // Highlight any search terms if we have them
+    // Highlight any search terms if we have them (except if they are inside html tags)
     if (isset($_GET['q']) && strlen($_GET['q'])) {
-        $content = preg_replace('/(' . $_GET['q'] . ')/i', "<mark>$1</mark>", $content);
+        $content = preg_replace(
+            '/<.*?' . $_GET['q'] . '.*?>(*SKIP)(*FAIL)|' . $_GET['q'] . '/i',
```

Comment:
WHAT BLACK MAGIC IS THIS?!

![](https://media.giphy.com/media/y4qa6akskDksylSBBn/giphy.gif)

Reply:
Found it on SO and I find it more read able then the lookahead a la ` (?=.*?<.*?/>)`

Reply:
If it works, then great! I've never seen it before.